doc/ChangeLog

```diff
@@ -0,0 +1,93 @@
+
+Code reviewed by: cacraig, peverwhee, mwaxmonsky
+
+List all files eliminated:
+
+utilities/geopotential_t.F90 - Replaced with utilities/geopotential_temp.F90
+utilities/geopotential_t.meta - Replaced with utilities/geopotential_temp.meta
+
+List all files added and what they do:
+
+utilities/geopotential_temp.F90 - New geopotential calculation routine which uses
+                                  generalized virtual temperature (i.e. all
+                                  thermodynamically active species).
+utilities/geopotential_temp.meta - Metdata file for new "geopotential_temp.F90" CCPP scheme.
+
+List all existing files that have been modified, and describe the changes:
+
+held_suarez/held_suarez_1994.F90         - Code cleanup
+held_suarez/held_suarez_1994.meta        - Updated variable standard names and removed unused variables
+kessler/kessler.F90                      - Code cleanup and bug fixes
+kessler/kessler.meta                     - Updated variable standard names and removed unused variables
+kessler/kessler_update.F90               - Code cleanup
+kessler/kessler_update.meta              - Updated variable standard names and did some metadata cleanup
+suite_held_suarez_1994.xml               - Updated scheme names, and modified scheme list to better match CAM.
+suite_kessler.xml                        - Updated scheme names.
+utilities/physics_tendency_updaters.F90  - Updated variable names and dimensioning
+utilities/physics_tendency_updaters.meta - Updated variable standard names
+utilities/qneg.F90                       - Code cleanup
+utilities/qneg.meta                      - Updated variable standard names and removed unused variables
+utilities/state_converters.F90           - Code cleanup and updated variable/scheme names.
+utilities/state_converters.meta          - Updated variable standard names and scheme names,
+                                           and removed unused variables.
+utilities/static_energy.F90              - Code cleanup and updated variable dimensioning
+utilities/static_energy.meta             - Updated variable standard names
+
+List all standard names currently not in the CCPPStandardNames dictionary:
+
+    - composition_dependent_specific_heat_of_dry_air_at_constant_pressure
+    - reference_pressure_in_atmosphere_layer_normalized_by_reference_pressure
+    - heating_rate
+    - tendency_of_northward_wind
+    - composition_dependent_ratio_of_dry_air_gas_constant_to_specific_heat_of_dry_air_at_constant_pressure
+    - tendency_of_eastward_wind
+    - scheme_name
+    - rain_mixing_ratio_wrt_dry_air
+    - density_of_dry_air
+    - density_of_fresh_liquid_water_at_0c
+    - composition_dependent_gas_constant_of_dry_air
+    - tendency_of_northward_wind_due_to_model_physics
+    - tendency_of_eastward_wind_due_to_model_physics
+    - number_of_ccpp_constituents
+    - flag_for_mpi_root
+    - ccpp_constituent_minimum_values
+    - log_output_unit
```

Comment:
Here is an idea (not to be implemented here). Right now, we have `ccpp_error_message` and `ccpp_error_code`. The framework only checks `ccpp_error_code` for non-zero status to decide if it needs to return early. It could also collect any content in `ccpp_error_message` when `ccpp_error_code == 0` and return all such output if the final `ccpp_error_code == 0`. Then the host model could log the output in such cases. If this is not obviously a terrible idea, I can help flesh out the concept in the CCPP Framework repo.

Reply:
We discussed this at the CCPP-framework meeting on Sept. 25, and it sounds like for now our method is fine (NOAA is apparently violating this rule all over the place), and that once the latest round of capgen PRs has been merged folks will start work on implementing a logging capability in the capgen branch.  I believe all discussion on this is supposed to happen here:

https://github.com/NCAR/ccpp-framework/issues/439

@gold2718  I went ahead and quoted your idea in that issue along with one concern I had.  Feel free to reply there if you have any other thoughts/opinions.  Thanks!